### PR TITLE
Feature/CLD-404

### PR DESCRIPTION
**Pull request by vitalykorolev**
The changes for [CLD-404](https://project.marklogic.com/jira/browse/CLD-404) include the following:

- Update Jenkinsfile to make it work with the new pipeline
- Add check for Jira ID
- Organize stage logic into functions
- Add server regression as a manual stage
- Add integration with Jira
- Add PR checklist for best practice reminder

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,127 +1,174 @@
...
+    return reviewState
+}
+
+def getServerPath(branchName) {
+    switch (branchName) {
+            case '10.1':
+            return 'rh7v-10-tst-bld-1.eng.marklogic.com/b10_1'
+    case '11.0':
+            return 'rh7v-i64-11-build/HEAD'
+            case '9.0':
```
Comment (twosdai) [inline Jenkinsfile:87]: My brain just broke on this line, can we remove the extra tab/4spaces.

File: Jenkinsfile [hunk 2]
```diff
@@ -131,38 +178,111 @@ pipeline{
...
-            mail bcc: '', body: "<b>Jenkins pipeline for ${env.JOB_NAME} <br>Build Number: ${env.BUILD_NUMBER} <br>${env.BUILD_URL}</b>", cc: '', charset: 'UTF-8', from: '', mimeType: 'text/html', replyTo: '', subject: "BUILD UNSTABLE: ${env.JOB_NAME} #${env.BUILD_NUMBER}", to: "${params.failEmail}";
-        }   
+        always {
+                sh '''
+                    cd src/centos
+                    rm -rf *.rpm
+                '''
+        }
+        success {
+            ResultNotification('BUILD SUCCESS ✅')
```
Comment (twosdai) [inline Jenkinsfile:279]: I'm petitioning it to be an animal instead of a checkmark. =P 

🐱  or 🐸  will do.

**Review by twosdai — COMMENTED** [review]
small comments.

File: Jenkinsfile (hunk 1 above)
Comment (vitalykorolev) [reply Jenkinsfile:87]: I need to fix my groovy linter. For some reason it keeps inserting extra spaces.

**Review by twosdai — APPROVED** [review]
LGTM

**Pull request merged**